src/main/java/uk/gov/hmcts/divorce/document/content/NoticeOfProceedingJointJudicialSeparationContent.java

```diff
@@ -0,0 +1,85 @@
+
+@Component
+public class NoticeOfProceedingJointJudicialSeparationContent {
+
+    public static final String JUDICIAL_SEPARATION_PROCEEDINGS = "judicialSeparationProceedings";
+    public static final String JUDICIAL_SEPARATION = "judicialSeparation";
+    public static final String REISSUED_DATE = "ressiuedDate";
+    public static final String MARRIED_TO_MORE_THAN_ONE_PERSON = "marriedToMoreThanOnePerson";
+    public static final String MARRIED_TO_MORE_THAN_ONE_PERSON_TEXT = "You must tell the court if you’ve been married to more than"
+        + " one person during this marriage.";
+    public static final String JUDICIAL_SEPARATION_PROCEEDINGS_SUBTEXT = "separation proceedings";
+    public static final String JUDICIAL_SEPARATION_SUBTEXT = "separation";
+    public static final String JUDICIAL = "judicial";
+
+    public static final String REISSUED_TEXT = "Reissued on: ";
+
+    @Autowired
+    private CommonContent commonContent;
+
+    @Autowired
+    private DocmosisCommonContent docmosisCommonContent;
+
+    public Map<String, Object> apply(final CaseData caseData,
+                                     final Long ccdCaseReference,
+                                     Applicant applicant,
+                                     Applicant partner) {
+
+        final LanguagePreference applicantLanguagePreference = applicant.getLanguagePreference();
+        Map<String, Object> templateContent = docmosisCommonContent.getBasicDocmosisTemplateContent(
+            applicantLanguagePreference);
+
+        templateContent.put(RELATION, commonContent.getPartner(caseData, partner, applicantLanguagePreference));
+        templateContent.put(CASE_REFERENCE, formatId(ccdCaseReference));
+        templateContent.put(FIRST_NAME, applicant.getFirstName());
+        templateContent.put(LAST_NAME, applicant.getLastName());
+        templateContent.put(ADDRESS, applicant.getPostalAddress());
+        templateContent.put(ISSUE_DATE, caseData.getApplication().getIssueDate().format(DATE_TIME_FORMATTER));
+
+        if (!WELSH.equals(applicantLanguagePreference)) {
+            getEnglishTemplateContent(templateContent, caseData);
+        }
+
+        return templateContent;
+    }
+
+    private void getEnglishTemplateContent(Map<String, Object> templateContent, CaseData caseData) {
+
+        if (null != caseData.getApplication().getReissueDate()) {
+            templateContent.put(REISSUED_DATE, REISSUED_TEXT + caseData.getApplication().getReissueDate().format(DATE_TIME_FORMATTER));
+        }
+
+        StringBuilder judicialSeparationProceedingsFinalText = new StringBuilder(JUDICIAL_SEPARATION_PROCEEDINGS_SUBTEXT);
+        StringBuilder judicialSeparationFinalText = new StringBuilder(JUDICIAL_SEPARATION_SUBTEXT);
+        if (caseData.isDivorce()) {
+            judicialSeparationProceedingsFinalText.insert(0, JUDICIAL + " ");
+            judicialSeparationFinalText.insert(0, JUDICIAL + " ");
```

Comment:
can we not simplify this it's bit tricky to understand ? may be use constant strings ?

Reply:
I did try that but sonar flagged it and said I needed to use a StringBuilder or StringBuffer and didn't like it when I just concatenated the strings

Reply:
is that a bit clearer?

Reply:
```   public static final String JUDICIAL = "judicial ";```